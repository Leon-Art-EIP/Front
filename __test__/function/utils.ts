import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
