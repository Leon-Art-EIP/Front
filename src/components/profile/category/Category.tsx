"use client";

import React from "react";
import Button from "../Button";

export type TCategory =
  | "Huile"
  | "Aquarelle"
  | "Acrylique"
  | "Gouache"
  | "Tempéra"
  | "Fresque"
  | "Crayon"
  | "Fusain"
  | "Encre"
  | "Pastel"
  | "Sanguine"
  | "Craie"
  | "Photographie argentique"
  | "Photographie numérique"
  | "Photographie noir et blanc"
  | "Photographe couleur"
  | "Bronze"
  | "Pierre"
  | "Bois"
  | "Résine"
  | "Céramique"
  | "Verre"
  | "Porcelaine"
  | "Faïence"
  | "Grès"
  | "Terre cuite"
  | "Broderie"
  | "Tapisserie"
  | "Art vestimentaire"
  | "Linogravure"
  | "Eau-forte"
  | "Lithographie"
  | "Sérigraphie"
  | "Monotype"
  | "Soufflage de verre"
  | "Vitrail"
  | "Fusing"
  | "Joaillerie"
  | "Ebénisterie"
  | "Marqueterie"
  | "Ferronnerie d'art";

export interface ICategoryProps {
  category: TCategory;
}

/* c8 ignore start */
export default function Category(props: ICategoryProps): JSX.Element {
  return (
    <Button
      onClick={() => {}}
      text={props.category}
      className="text-[#4A4A4A] font-semibold bg-white"
    />
  );
}
/* c8 ignore end */
