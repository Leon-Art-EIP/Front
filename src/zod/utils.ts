export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
