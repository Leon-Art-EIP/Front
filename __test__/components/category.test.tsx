import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Category from "../../src/components/profile/category/Category";

test("Category snapshot", () => {
  testSnapshot(<Category category="Acrylique" />);
});
