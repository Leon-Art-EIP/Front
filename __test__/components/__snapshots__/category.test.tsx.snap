// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Category snapshot 1`] = `
<button
  className="inline-flex py-0.5 px-5 text-center justify-center text-xs rounded-2xl border-none self-center text-[#4A4A4A] font-semibold bg-white"
  onClick={[Function]}
>
  Acrylique
</button>
`;
